python/tutorials/06-fused-attention.py

```diff
@@ -569,9 +569,10 @@ def test_op(Z, H, N_CTX, D_HEAD, causal, dtype=torch.float16):
     tri_dq, q.grad = q.grad.clone(), None
     # compare
     assert torch.allclose(ref_out, tri_out, atol=1e-2, rtol=0)
-    assert torch.allclose(ref_dv, tri_dv, atol=1e-2, rtol=0)
-    assert torch.allclose(ref_dk, tri_dk, atol=1e-2, rtol=0)
-    assert torch.allclose(ref_dq, tri_dq, atol=1e-2, rtol=0)
+    rtol = 0.0 if torch.version.hip is None else 1e-2
```

Comment:
Absolute error looks like this: 
![Screenshot from 2024-04-12 23-48-35](https://github.com/openai/triton/assets/3631351/eede2be7-f9f9-4e67-9e82-c5bf4b935797)

Unfortunately at this point I can not answer why error is larger.

Reply:
I think it would make sense to investigate the larger error first. This tutorial is running fp16 and it should match nvidia. I don't think we are in a rush to merge this, given that it wouldn't be helpful for testing internal workloads due to the increased rtol.

Reply:
I agree with Zahi. Since it's all running IEEE FP16, I can't think of a reason why the error should be larger if denorms are properly handled.

Reply:
Yea this makes sense to me, thanks